index.js

```diff
@@ -62,6 +62,32 @@ function Warehouse(options) {
   this.releaseLine = new Warehouse.ReleaseLine(this, options.releaseLine);
 }
 
+/**
+ * Normalize possible auth arguments
+ * @function _normalizeAuth
+ * @param {Object|String} opts - Auth options from constructor
+ * @returns {String} to be used for authorization header
+ */
+Warehouse.prototype._normalizeAuth = function _normalizeAuth(opts) {
+  if (!opts) return;
+  // Legacy support is if opts is a string, we send it as a bearer token
+  if (typeof opts === 'string') return 'Bearer ' + opts;
+
+  if (!opts.type || !opts.token) return;
```

Comment:
Like @msluther mentioned below, we should probably throw/warn/some kind of feedback here.